scripts/download.py

```diff
@@ -26,15 +31,38 @@ def download_from_hub(repo_id: Optional[str] = None, access_token: Optional[str]
             " https://huggingface.co/settings/tokens"
         )
 
+    download_files = ["tokenizer*", "generation_config.json"]
+    if download_safetensors:
+        from safetensors.torch import load_file as safetensors_load
+        download_files.extend("*.safetensors*")
+    else:
+        download_files.extend("*.bin*")
+
     snapshot_download(
         repo_id,
         local_dir=f"checkpoints/{repo_id}",
         local_dir_use_symlinks=False,
         resume_download=True,
-        allow_patterns=["*.bin*", "tokenizer*", "generation_config.json"],
+        allow_patterns=download_files,
         token=access_token,
     )
 
+    # convert safetensors to PyTorch binaries
+    if download_safetensors:
+        print("Converting .safetensor files to PyTorch binaries (.bin)")
+        directory = f"checkpoints/{repo_id}"
+        for filename in os.listdir(directory):
+            full_path = os.path.join(directory, filename)
+
+            if ".safetensors" in filename:
+                new_filename = filename.replace('.safetensors', '.bin')
+                new_filename = "pytorch_" + new_filename
+                new_full_path = os.path.join(directory, new_filename)
+
+                print(f"{filename} --> {new_filename}")
+                pt_state_dict = safetensors_load(full_path)
+                torch.save(pt_state_dict, new_full_path)
```

Comment:
This should also remove the safetensor file IMO. Or this will use a lot of disk

Reply:
Yes, I was thinking about this was well, but I wanted to leave it first in case there's some issue. It would be frustrating if you have to redownload. But yeah, I agree.